### ISSUE TITLE
Low precision model prediction 

### ISSUE THREAD

**Issue by eric-haibin-lin** [eric-haibin-lin/mxnet#17]
Experiment Setup 
- Baseline 

Trained [Inception BN Model(small)](https://github.com/dmlc/mxnet/blob/master/example/image-classification/symbol_inception-bn-28-small.py) on cifar-10 dataset with 20 epochs. 

The validation accuracy of this model is: 
2016-07-04 20:47:55,781 Node[0] Epoch[20] Validation-accuracy=0.881230
2016-07-04 20:47:55,781 Node[0] Epoch[20] Validation-top_k_accuracy_5=0.995154
2016-07-04 20:47:55,781 Node[0] Epoch[20] Validation-top_k_accuracy_10=1.000000
2016-07-04 20:47:55,781 Node[0] Epoch[20] Validation-top_k_accuracy_20=1.000000
- Rounded model

Just to kick it started, use the trained model for prediction, except that all the intermediate results in between operators are [**rounded** to closest integers](https://github.com/eric-haibin-lin/mxnet/blob/master/example/image-classification/symbol_inception-bn-28-small.py#L23). There's no rounding during training. No other quantization method is applied yet. 

The validation accuracy of the rounded model is: 
2016-07-05 02:23:44,401 Node[0] Epoch[20] Validation-accuracy=0.698081
2016-07-05 02:23:44,401 Node[0] Epoch[20] Validation-top_k_accuracy_5=0.969244
2016-07-05 02:23:44,401 Node[0] Epoch[20] Validation-top_k_accuracy_10=1.000000
2016-07-05 02:23:44,401 Node[0] Epoch[20] Validation-top_k_accuracy_20=1.000000

@harouwu  @muli
